### PR TITLE
DROOLS-2741: [DMN Designer] Clear the diagram toolbar button

### PR DESCRIPTION
**Pull request by manstis**
This PR covers both:-

- https://issues.jboss.org/browse/DROOLS-2741
- https://issues.jboss.org/browse/DROOLS-2742

This PR requires on https://github.com/kiegroup/kie-wb-common/pull/1975 being merged. After that PR is merged I will rebase this PR to include only the changes for the caption JIRA (at the moment only the last 2 commits relate to the JIRAs).

### CODE REVIEW

**Comment by jomarko** [comment]
@manstis I will do review once the #1975 is in to be sure I build locally the right source code.

**Comment by manstis** [comment]
@jomarko This is good to review now. Thanks.

File: kie-wb-common-dmn/kie-wb-common-dmn-project-client/src/main/java/org/kie/workbench/common/dmn/project/client/editor/ProjectToolbarStateHandler.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            projectEditorMenuSessionItems.setItemEnabled(command, false);
+        });
+    }
+
+    @Override
+    public void enterGraphView() {
+        commandStates.entrySet().forEach(entry -> {
+            final Class<? extends ClientSessionCommand> command = entry.getKey();
+            projectEditorMenuSessionItems.setItemEnabled(command, entry.getValue());
+        });
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-project-client/src/main/java/org/kie/workbench/common/dmn/project/client/editor/ProjectToolbarStateHandler.java]: Question, I don't see the added value of `commandStates`. Can't we implement methods just like below?
```java
   @Override
    public void enterGridView() {
        Arrays.stream(COMMAND_CLASSES)
                .forEach(toolbarCommand -> projectEditorMenuSessionItems.setItemEnabled(toolbarCommand, false));
    }

    @Override
    public void enterGraphView() {
        Arrays.stream(COMMAND_CLASSES)
                .forEach(toolbarCommand -> projectEditorMenuSessionItems.setItemEnabled(toolbarCommand, true));
    }
```

File: kie-wb-common-dmn/kie-wb-common-dmn-webapp/src/main/java/org/kie/workbench/common/dmn/showcase/client/screens/editor/StandaloneToolbarStateHandler.java [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    private boolean undoToolbarCommandEnabled = false;
+    private boolean redoToolbarCommandEnabled = false;
+    private boolean validateToolbarCommandEnabled = false;
+    private boolean exportToPngToolbarCommandEnabled = false;
+    private boolean exportToJpgToolbarCommandEnabled = false;
+    private boolean exportToPdfToolbarCommandEnabled = false;
+    private boolean copyCommandEnabled = false;
+    private boolean cutCommandEnabled = false;
+    private boolean pasteCommandEnabled = false;
+    private boolean saveCommandEnabled = false;
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-webapp/src/main/java/org/kie/workbench/common/dmn/showcase/client/screens/editor/StandaloneToolbarStateHandler.java]: Similar question here, do we need to store the state of particular toolbar buttons?

File: kie-wb-common-dmn/kie-wb-common-dmn-project-client/src/main/java/org/kie/workbench/common/dmn/project/client/editor/ProjectToolbarStateHandler.java (hunk 1 above)
Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-project-client/src/main/java/org/kie/workbench/common/dmn/project/client/editor/ProjectToolbarStateHandler.java]: @jomarko Can you *guarantee* that all the commands were *enabled* before they are disabled? I store their current state before disabling to simply ensure they are restored to their original state (without me having to trawl the entire code base to guarantee they will always be enabled - e.g. what about a read-only session when opening an older version).

File: kie-wb-common-dmn/kie-wb-common-dmn-webapp/src/main/java/org/kie/workbench/common/dmn/showcase/client/screens/editor/StandaloneToolbarStateHandler.java (hunk 2 above)
Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-webapp/src/main/java/org/kie/workbench/common/dmn/showcase/client/screens/editor/StandaloneToolbarStateHandler.java]: @jomarko As above.

File: kie-wb-common-dmn/kie-wb-common-dmn-project-client/src/main/java/org/kie/workbench/common/dmn/project/client/editor/ProjectToolbarStateHandler.java (hunk 1 above)
Comment (jomarko) [reply kie-wb-common-dmn/kie-wb-common-dmn-project-client/src/main/java/org/kie/workbench/common/dmn/project/client/editor/ProjectToolbarStateHandler.java]: @manstis, great, I was not assuming readonly session. Will recheck and let you know.Thanks!

Comment (jomarko) [reply kie-wb-common-dmn/kie-wb-common-dmn-project-client/src/main/java/org/kie/workbench/common/dmn/project/client/editor/ProjectToolbarStateHandler.java]: @manstis found this during re-review https://issues.jboss.org/browse/DROOLS-2765. But not sure if related or not.

Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-project-client/src/main/java/org/kie/workbench/common/dmn/project/client/editor/ProjectToolbarStateHandler.java]: @jomarko I have commented the new JIRA. It is unrelated to this PR.

**Pull request merged**